image_tools_py/src/image_qos_profile.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from rclpy.qos import QoSDurabilityPolicy, QoSHistoryPolicy, QoSReliabilityPolicy
+from rclpy.qos import qos_profile_default
+
+
+def create_qos_profile(depth, reliability_policy, keep):
+    qos_profile = qos_profile_default
+
+    # Depth represents how many messages to store in history when the history
+    # policy is KEEP_LAST.
+    qos_profile.depth = depth
+
+    # The reliability policy can be reliable, meaning that the underlying
+    # transport layer will try to ensure that every message gets received in
+    # order, or best effort, meaning that the transport makes no guarantees about
+    # the order or reliability of delivery.
+    if reliability_policy == 0:
+        qos_profile.reliability = QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT
+    elif reliability_policy == 1:
+        qos_profile.reliability = QoSReliabilityPolicy.RMW_QOS_POLICY_RELIABILITY_RELIABLE
+    else:
+        raise Exception("Invalid reliability policy (must be 0 for best effort or 1 for reliable")
+
+    # The history policy determins how messages are saved until the message is
+    # taken by the reader.
+    # KEEP_ALL saves all messages until they are taken.
+    # KEEP_LAST enforces a limit on the number of messages that are saved,
+    # specified by the 'depth' parameter.
+    if keep == 0:
+        qos_profile.history = QoSHistoryPolicy.RMW_QOS_POLICY_HISTORY_KEEP_LAST
+    elif keep == 1:
+        qos_profile.history = QoSHistoryPolicy.RMW_QOS_POLICY_HISTORY_KEEP_ALL
+    else:
+        raise Exception("Invalid keep policy (must be 0 for keep last or 1 for keep all")
+
+    qos_profile.durability = QoSDurabilityPolicy.RMW_QOS_POLICY_DURABILITY_VOLATILE
```

Comment:
durability is already volatile in `qos_profile_default` so this line can be removed.